micrometer-core/src/main/java/io/micrometer/core/instrument/push/PushMeterRegistry.java

```diff
@@ -71,9 +76,12 @@ public void start(ThreadFactory threadFactory) {
             logger.info("publishing metrics for " + this.getClass().getSimpleName() + " every " + TimeUtils.format(config.step()));
 
             scheduledExecutorService = Executors.newSingleThreadScheduledExecutor(threadFactory);
-            // time publication to happen just after StepValue finishes the step
+            // if aligned, time publication to happen just after StepValue finishes the step
             long stepMillis = config.step().toMillis();
-            long initialDelayMillis = stepMillis - (clock.wallTime() % stepMillis) + 1;
+            long initialDelayMillis = stepMillis;
+            if (config.stepAlignment()) {
+                initialDelayMillis -= (clock.wallTime() % stepMillis) + 1;
```

Comment:
I believe this will end up ultimately subtracting the +1 at the end o_O. Unless this is intended ?

Reply:
heh, good catch! It should be 1 ms more